components/authz-service/storage/postgres/migration/sql/47_staged_rules_tables.up.sql

```diff
@@ -0,0 +1,76 @@
+BEGIN;
+
+CREATE TABLE iam_staged_project_rules (
+  db_id SERIAL PRIMARY KEY,
+  id TEXT NOT NULL UNIQUE,
+  project_id TEXT REFERENCES iam_projects ON DELETE CASCADE,
+  name TEXT NOT NULL,
+  type TEXT NOT NULL,
+  deleted BOOLEAN NOT NULL DEFAULT FALSE
+);
```

Comment:
I know we've discussed this to a large extent, and I'm a jerk for bringing this up now -- but I need the DDL to _think RDBMS_ 😅 

1. These two tables are basically copies of the other ones, right? With an extra field.

   I think that means [we could use something like this](https://dba.stackexchange.com/a/207103):
   ```sql
   CREATE TABLE iam_staged_project_rules ( 
     deleted BOOLEAN NOT NULL DEFAULT FALSE, 
     LIKE iam_project_rules INCLUDING ALL
   );
   ```
2. We didn't want to add a column with some enum -- like `deleted, staged, active` -- to the other ones... would anyone mind iterating the reason for that for me once more? 😉

Reply:
Oh, yes, #1 is nicer. I think we were underwater considering whether that would copy data, too. 👍 

I'll draw up a document outlining pros/cons of our choices and what seems to have led us here. I don't quite know what you mean by " to the other ones," though. Could you explain?

Reply:
Ooooh didn't know about the `LIKE` keyword. That's awesome.

Reply:
Ok, turns out that while the `LIKE` is prettier, the tables have a few distinctions which make maintaining the foreign key constraints a little less straightforward. 

`iam_staged_project_rules` references `iam_projects`, while `iam_staged_rule_conditions` references `iam_staged_project_rules`. We don't yet need a staging table for projects (though we might), so I think it's more straightforward to leave it like this.